### ISSUE TITLE
Using the traditional SQL "group by" syntax returns inconsistent and duplicate data

### ISSUE THREAD

**Issue by nerdyness** [influxdb/influxdb#2583]
Hi all,

Forgive me if this has been reported before - I looked through existing issues but couldn't find it.

I have a series "series" which has amongst other columns a text column called "incident_key". I'd like to know how many incident_keys were inserted into influxdb per hour and fill non-existing values with 0. The following query works perfectly for a small set of data:

```
$ curl -G 'http://influxdb:8086/db/mydb/series?u=user&p=pass' --data-urlencode 'q=select count(incident_key) from series where time > 1431618600s and time < 1431662482s group by time(1h) fill(0)'
[{"name":"series","columns":["time","count"],"points":[[1431662400000,0],[1431658800000,8],[1431655200000,2],[1431651600000,4],[1431648000000,2],[1431644400000,0],[1431640800000,0],[1431637200000,0],[1431633600000,2],[1431630000000,1],[1431626400000,3],[1431622800000,2],[1431619200000,4],[1431615600000,0]]}]
```

If I increase the date range, influxdb returns each point twice. Once with the actual value and then again with a count of 0 (or whatever "fill" is set to).

```
$ curl -G 'http://influxdb:8086/db/mydb/series?u=user&p=pass' --data-urlencode 'q=select count(incident_key) from series where time > 1431525600s and time < 1431662482s group by time(1h) fill(0)'
[{"name":"series","columns":["time","count"],"points":[[1431662400000,0],[1431658800000,8],[1431655200000,2],[1431651600000,4],[1431648000000,2],[1431644400000,0],[1431640800000,0],[1431637200000,0],[1431633600000,2],[1431630000000,1],[1431626400000,3],[1431622800000,2],[1431619200000,4],[1431615600000,0],[1431612000000,0],[1431608400000,2],[1431604800000,2],[1431601200000,5],[1431597600000,1],[1431594000000,2],[1431590400000,5],[1431586800000,0],[1431583200000,3],[1431579600000,1],[1431576000000,17],[1431572400000,4],[1431568800000,9],[1431565200000,19],[1431561600000,18],[1431558000000,0],[1431554400000,0],[1431550800000,0],[1431547200000,0],[1431543600000,0],[1431540000000,0],[1431536400000,0],[1431532800000,0],[1431529200000,0],[1431525600000,0],[1431662400000,0],[1431658800000,0],[1431655200000,0],[1431651600000,0],[1431648000000,0],[1431644400000,0],[1431640800000,0],[1431637200000,0],[1431633600000,0],[1431630000000,0],[1431626400000,0],[1431622800000,0],[1431619200000,0],[1431615600000,0],[1431612000000,0],[1431608400000,0],[1431604800000,0],[1431601200000,0],[1431597600000,0],[1431594000000,0],[1431590400000,0],[1431586800000,0],[1431583200000,0],[1431579600000,0],[1431576000000,0],[1431572400000,0],[1431568800000,0],[1431565200000,0],[1431561600000,0],[1431558000000,0],[1431554400000,0],[1431550800000,0],[1431547200000,3],[1431543600000,1],[1431540000000,3],[1431536400000,17],[1431532800000,2],[1431529200000,4],[1431525600000,3]]}]
```

However if I restructure my query as per some of the examples on [query language](http://influxdb.com/docs/v0.8/api/query_language.html) I get the correct data back:

```
$ curl -G 'http://influxdb:8086/db/mydb/series?u=user&p=pass' --data-urlencode 'q=select count(incident_key) from series group by time(1h), fill(0) where time > 1431525600s and time < 1431662421s'
[{"name":"series","columns":["time","count"],"points":[[1431658800000,8],[1431655200000,2],[1431651600000,4],[1431648000000,2],[1431633600000,2],[1431630000000,1],[1431626400000,3],[1431622800000,2],[1431619200000,4],[1431608400000,2],[1431604800000,2],[1431601200000,5],[1431597600000,1],[1431594000000,2],[1431590400000,5],[1431583200000,3],[1431579600000,1],[1431576000000,17],[1431572400000,4],[1431568800000,9],[1431565200000,19],[1431561600000,18],[1431547200000,3],[1431543600000,1],[1431540000000,3],[1431536400000,17],[1431532800000,2],[1431529200000,4],[1431525600000,3]]}]
```

This version of influxdb was installed using the normal package management system of my distribution:

```
$ aptitude show influxdb
Package: influxdb
State: installed
Automatically installed: no
Version: 0.8.8
Priority: extra
Section: default
Maintainer: <root@release>
Architecture: amd64
Uncompressed Size: 54.2 M
Description: no description given

Homepage: http://example.com/no-uri-given
```

Which is:

```
$ uname -a
Linux host 3.13.0-36-generic #63-Ubuntu SMP Wed Sep 3 21:30:07 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:  Ubuntu 14.04.2 LTS
Release:  14.04
Codename: trusty
```

And here is the data in 'series':

```
$ curl -G 'http://influxdb:8086/db/mydb/series?u=user&p=pass' --data-urlencode 'q=select incident_key from series where time > 1431525600s and time < 1431662482s'
[{"name":"series","columns":["time","sequence_number","incident_key"],"points":[[1431661879000,4936450001,"text"],[1431661414000,4936460001,"text"],[1431660334000,4936400001,"text"],[1431660253000,4936410001,"text"],[1431660193000,4936420001,"text"],[1431660184000,4936430001,"text"],[1431660124000,4936440001,"text"],[1431658808000,4936390001,"text"],[1431657778000,4936380001,"text"],[1431657203000,4936370001,"text"],[1431654954000,4936360001,"text"],[1431654544000,4936340001,"text"],[1431654454000,4936350001,"text"],[1431652517000,4936330001,"text"],[1431651596000,4936320001,"text"],[1431651034000,4936310001,"text"],[1431633877000,4936190001,"text"],[1431633646000,4936200001,"text"],[1431632347000,4936210001,"text"],[1431629046000,4936220001,"text"],[1431628845000,4936230001,"text"],[1431627867000,4936240001,"text"],[1431626016000,4936250001,"text"],[1431624102000,4936260001,"text"],[1431622272000,4936270001,"text"],[1431621646000,4936280001,"text"],[1431620511000,4936290001,"text"],[1431620145000,4936300001,"text"],[1431611663000,4936170001,"text"],[1431610643000,4936160001,"text"],[1431605948000,4936150001,"text"],[1431604845000,4936140001,"text"],[1431602917000,4936120001,"text"],[1431602871000,4936130001,"text"],[1431602433000,4936110001,"text"],[1431601681000,4936090001,"text"],[1431601677000,4936100001,"text"],[1431597795000,4936080001,"text"],[1431594714000,4936070001,"text"],[1431594051000,4936060001,"text"],[1431592731000,4936040001,"text"],[1431592371000,4936050001,"text"],[1431592235000,4936020001,"text"],[1431592071000,4936030001,"text"],[1431590745000,4936010001,"text"],[1431585185000,4936000001,"text"],[1431585115000,4935990001,"text"],[1431583238000,4935980001,"text"],[1431581119000,4935970001,"text"],[1431578719000,4935930001,"text"],[1431578605000,4935940001,"text"],[1431578596000,4935950001,"text"],[1431578575000,4935960001,"text"],[1431578455000,4935870001,"text"],[1431578445000,4935880001,"text"],[1431578428000,4935890001,"text"],[1431578426000,4935900001,"text"],[1431578415000,4935910001,"text"],[1431578335000,4935920001,"text"],[1431577804000,4935860001,"text"],[1431576925000,4935800001,"text"],[1431576889000,4935810001,"text"],[1431576885000,4935820001,"text"],[1431576865000,4935830001,"text"],[1431576814000,4935840001,"text"],[1431576775000,4935850001,"text"],[1431575907000,4935770001,"text"],[1431575644000,4935780001,"text"],[1431575571000,4935790001,"text"],[1431574924000,4935760001,"text"],[1431571773000,4935700001,"text"],[1431571746000,4935710001,"text"],[1431571725000,4935720001,"text"],[1431571495000,4935730001,"text"],[1431571354000,4935740001,"text"],[1431571343000,4935750001,"text"],[1431571253000,4935670001,"text"],[1431571225000,4935680001,"text"],[1431571175000,4935690001,"text"],[1431568397000,4935660001,"text"],[1431568311000,4935560001,"text"],[1431568245000,4935570001,"text"],[1431568235000,4935580001,"text"],[1431568199000,4935590001,"text"],[1431568197000,4935600001,"text"],[1431568157000,4935610001,"text"],[1431568135000,4935620001,"text"],[1431568065000,4935630001,"text"],[1431568005000,4935640001,"text"],[1431567876000,4935650001,"text"],[1431567273000,4935540001,"text"],[1431567156000,4935550001,"text"],[1431567115000,4935500001,"text"],[1431567025000,4935510001,"text"],[1431566927000,4935520001,"text"],[1431566695000,4935530001,"text"],[1431566185000,4935480001,"text"],[1431566119000,4935490001,"text"],[1431565037000,4935440001,"text"],[1431565037000,4935430001,"text"],[1431564495000,4935420001,"text"],[1431563388000,4935400001,"text"],[1431563185000,4935410001,"text"],[1431562819000,4935340001,"text"],[1431562807000,4935350001,"text"],[1431562667000,4935360001,"text"],[1431562363000,4935380001,"text"],[1431562324000,4935390001,"text"],[1431562287000,4935260001,"text"],[1431562217000,4935270001,"text"],[1431562187000,4935280001,"text"],[1431562157000,4935290001,"text"],[1431562125000,4935300001,"text"],[1431562075000,4935310001,"text"],[1431561913000,4935330001,"text"],[1431561685000,4935250001,"text"],[1431549465000,4934850001,"text"],[1431548057000,4934860001,"text"],[1431547489000,4934870001,"text"],[1431546209000,4934880001,"text"],[1431541895000,4934890001,"text"],[1431540949000,4934900001,"text"],[1431540913000,4934910001,"text"],[1431539430000,4934920001,"text"],[1431539406000,4934930001,"text"],[1431539399000,4934940001,"text"],[1431538415000,4934950001,"text"],[1431538296000,4934960001,"text"],[1431538279000,4934970001,"text"],[1431538207000,4934980001,"text"],[1431538185000,4934990001,"text"],[1431538176000,4935000001,"text"],[1431538125000,4935010001,"text"],[1431538099000,4935020001,"text"],[1431538096000,4935030001,"text"],[1431537996000,4935040001,"text"],[1431537882000,4935050001,"text"],[1431537239000,4935060001,"text"],[1431536507000,4935070001,"text"],[1431536505000,4935080001,"text"],[1431535869000,4935090001,"text"],[1431535497000,4935130001,"text"],[1431532249000,4935140001,"text"],[1431531811000,4935180001,"text"],[1431531189000,4935190001,"text"],[1431531087000,4935200001,"text"],[1431528989000,4935210001,"text"],[1431526289000,4934830001,"text"],[1431525682000,4934770001,"text"]]}]
```

Let me know if you need any further input from me.

**Comment by toddboom** [comment]
@nerdyness there were some issues with v0.8.x and duplicate data when crossing shard boundaries, which sounds like it might explain the behavior you're seeing. unfortunately because v0.9.0 is going to be released soon, we're not going be releasing any additional versions for the v0.8.x family. this behavior should be fixed on the v0.9.0 RCs.

**Comment by nerdyness** [comment]
Thanks @toddboom. I've simply adopted the syntax that works for me - this issue was meant to bring this to your attention in case you weren't aware.
